### PR TITLE
[WFLY-12394] Execute portions of the testsuite using a slimmed server instead of the normal dist

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.jboss.org/browse/WFLY-12394

This would be accompanied by new PR jobs that run with -Dts.layers rather than -DallTests which in turn would trigger slimmed installations and different surefire executions in the various testsuites.

@jfdenise @mjurc @yersan @jmesnil  FYI.

### CODE REVIEW

**Comment by bstansberry** [comment]
A couple custom jobs meant to look like the new PR jobs I contemplate:

https://ci.wildfly.org/viewLog.html?buildId=162612&tab=buildResultsDiv&buildTypeId=WF_MasterLinux

https://ci.wildfly.org/viewLog.html?buildId=162600&tab=buildResultsDiv&buildTypeId=WF_MasterWindowsJdk11

I'm not sure if those are running correctly at the moment; will check again later.

**Comment by rsvoboda** [comment]
@msve fyi

File: testsuite/integration/microprofile-tck/config/pom.xml [hunk 1]
```diff
@@ -79,4 +79,103 @@
...
+                                            <inherit-packages>false</inherit-packages>
+                                        </feature-pack>
+                                    </feature-packs>
+                                    <configurations>
+                                        <config>
+                                            <model>standalone</model>
+                                            <name>standalone.xml</name>
+                                            <layers>
+                                                <layer>core-server</layer>
+                                                <layer>observability</layer>
```
Comment (jfdenise) [inline testsuite/integration/microprofile-tck/config/pom.xml]: @bstansberry , you are expecting observability to bring web-server through its dependency on cdi or you don't need webserver?
When we have WFLY-12299 in, web-server would have to be added explicitly. Same apply to the other provisioned server.

Comment (bstansberry) [reply testsuite/integration/microprofile-tck/config/pom.xml]: Yes, I was counting on it being pulled transitively. I'll update that.

Comment (bstansberry) [reply testsuite/integration/microprofile-tck/config/pom.xml]: Done.

**Comment by bstansberry** [comment]
I've taken this out of draft state and removed the hold label as this is good for review.

There's more I plan to do on this topic but this is a large and useful chunk.

**Review by jfdenise — APPROVED** [review]

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**